uitools/common/src/ArcGISAuthenticationController.cpp

```diff
@@ -187,13 +187,19 @@ void ArcGISAuthenticationController::handleNetworkAuthenticationChallenge(Networ
  */
 void ArcGISAuthenticationController::continueWithServerTrust(bool trust)
 {
-  // to simplify the toolkit experience, 'remember' is always true.
-  // it's uncommon to reject, and want another opportunity to accept later for the same host.
-  constexpr auto remember = true;
-  auto* credential = ServerTrustCredential::createWithChallenge(m_currentNetworkChallenge.get(), trust, remember);
-  if (credential)
+  if (trust)
   {
-    m_currentNetworkChallenge->continueWithCredential(credential);
+    auto* credential = ServerTrustCredential::createWithChallenge(m_currentNetworkChallenge.get(), this);
+    if (credential)
+    {
+      m_currentNetworkChallenge->continueWithCredential(credential);
+      m_currentNetworkChallenge.reset();
+    }
```

Comment:
if no credential, m_currentNetworkChallenge is not reset ?

Reply:
You are right, that is not covered. In theory, `ServerTrustCredential::createWithChallenge` could return null if the challenge type doesn't match, so I covered that possibility.

Good catch, thanks.